gazebo/Server.cc

```diff
@@ -582,16 +583,28 @@ void Server::Run()
 
   this->dataPtr->initialized = true;
 
+  IGN_PROFILE_THREAD_NAME("gzserver");
   // Stay on this loop until Gazebo needs to be shut down
   // The server and sensor manager outlive worlds
   while (!this->dataPtr->stop)
   {
+    IGN_PROFILE("Server::Run");
+    IGN_PROFILE_BEGIN("ProcessControlMsgs");
     this->ProcessControlMsgs();
+    IGN_PROFILE_END();
 
     if (physics::worlds_running())
+    {
+      IGN_PROFILE_BEGIN("run_once");
       sensors::run_once();
+      IGN_PROFILE_END();
+    }
     else if (sensors::running())
+    {
+      IGN_PROFILE_BEGIN("stop");
       sensors::stop();
+      IGN_PROFILE_END();
+    }
 
     common::Time::MSleep(1);
```

Comment:
hmm...having a 1 ms sleep in the server runloop seems non-ideal

Reply:
I remember seeing this before and wondering why it's there :monocle_face:

Reply:
I saw this while working on lockstepping. I tried removing the line but it caused some sensor tests to fail so I had to put it back in:
https://github.com/osrf/gazebo/pull/2746/files#diff-2e1da9464c22259ef9d9919412c42d2fR617

Reply:
I wonder if we can we change it to `NSleep(1)`?

this should happen in a separate PR; I'll try creating a PR with this change